core/src/main/java/com/nerds/gamejam/ecs/system/CameraSystem.java

```diff
@@ -6,24 +6,35 @@
 import com.badlogic.gdx.Input;
 import com.badlogic.gdx.graphics.OrthographicCamera;
 import com.badlogic.gdx.math.MathUtils;
+import com.badlogic.gdx.utils.viewport.FillViewport;
+import com.badlogic.gdx.utils.viewport.ScalingViewport;
 import com.nerds.gamejam.GameJam;
 import com.nerds.gamejam.util.InputUtil;
 
 @Wire
 public class CameraSystem extends BaseSystem {
 
     OrthographicCamera camera;
+    ScalingViewport scalingViewport;
 
     @Override
     protected void initialize() {
-        this.camera = new OrthographicCamera(512, GameJam.PLANET_VIEW_HEIGHT);
-        this.camera.zoom = 0.5f;
-        setCameraBounds();
+        this.camera = new OrthographicCamera();
+        this.camera.setToOrtho(false);
+        this.camera.zoom = 1;
+        float w = Gdx.graphics.getWidth();
+        float h = Gdx.graphics.getHeight();
+        float worldWidth = GameJam.PLANET_VIEW_HEIGHT * w / h;
+        int worldHeight = GameJam.PLANET_VIEW_HEIGHT;
+        scalingViewport = new FillViewport(worldWidth, worldHeight, camera);
+        scalingViewport.setScreenBounds(0, 0, Gdx.graphics.getWidth(), Gdx.graphics.getHeight());
+        scalingViewport.apply();
```

Comment:
I don't think the camera should be concerning itself with viewports

Reply:
Is this more at home in the bootstrap system instead?

Reply:
I think probably render system as the gui one can then set its own since render system is really the planet game world. The viewport shouldn't have had anything to do with camera buggy ness though. Viewport is just doing scaling etc of the viewable area eg getting rid of the black borders

Reply:
Moved this to the render system. Camera buggyness was probably a bad choice of words, it was more just not behaving the way I'd expect when doing positioning, like I couldn't centre the camera without doing weird resolution maths